contracts/STYLE_GUIDE.md

```diff
@@ -0,0 +1,347 @@
+
+This guide is not meant to be applied retroactively. There is no need to rewrite existing code to adhere to this guide, and when making (small) changes in existing files, it is not required to do so in accordance to this guide if it would conflict with other practices in that existing file. Consistency is preferred.
+
+We will be looking into `forge fmt`, but for now we still use `prettier`.
+
+# Code Organization
+
+- Group functionality together. E.g. Declare structs, events, and helper functions near the functions that use them. This is helpful when reading code because the related pieces are localized. It is also consistent with inheritance and libraries, which are separate pieces of code designed for a specific goal.
+- 🤔Why not follow the Solidity recommendation of grouping by visibility? Visibility is clearly defined next to the method signature, making it trivial to check. However, searching can be deceiving because of inherited methods. Given this inconsistency in grouping, we find it easier to read and more consistent to organize code around functionality. Additionally, we recommend testing the public interface for any Solidity contract to ensure it only exposes expected methods.
+- Follow the [Solidity folder structure CLIP](https://github.com/smartcontractkit/CLIPs/tree/main/clips/2023-04-13-solidity-folder-structure)
+
+### Comments
+
+- Comments should be in the `//` (default) or `///` (natspec) format, not the `/*  */` format.
+- Comments should follow [NatSpec](https://docs.soliditylang.org/en/latest/natspec-format.html)
+- Besides comment above functions/structs, comments should live everywhere a reader might be confused. Don’t overestimate the reader of your contract, expect confusion in many places and document accordingly. This will help massively during audits and onboarding new team members.
+- Headers should be used to group functionality, the following header style and length is recommended.
+  - Don’t use headers for a single function, or to say “getters”. Group by functionality e.g. the `Tokens and pools` , or `fees` logic within the CCIP OnRamp.
+
+```solidity
+  // ================================================================
+  // │                      Tokens and pools                        │
+  // ================================================================
+
+....
+
+  // ================================================================
+  // │                             Fees                             │
+  // ================================================================
+```
+
+### Imports
+
+- Imports should always be explicit
+- Imports should follow the following format:
+
+```solidity
+import {IInterface} from "../interfaces/IInterface.sol";
+
+import {AnythingElse} from "../code/AnythingElse.sol";
+
+import {ThirdPartyCode} from "../../vendor/ThirdPartyCode.sol";
```

Comment:
Is there a reason for this particular spacing?

Reply:
These three groups indicate very different usage: interfaces are a black box import, only caring about external functions and not whats inside. Third party imports should imo always be clearly separated from the rest because of the fact that it isn't our code you're working with. That leaves "AnythingElse" in the middle, which is not black box and not third party, so it implies a closer relationship